runtime/gc_stats.c

```diff
@@ -106,23 +104,31 @@ void caml_orphan_alloc_stats(caml_domain_state *domain) {
 /* The "sampled stats" of a domain are a recent copy of its
    domain-local stats, accessed without synchronization and only
    updated ("sampled") during stop-the-world events -- each minor
-   collection, and on domain termination. */
+   collection, including those happening during domain termination. */
 static struct gc_stats sampled_gc_stats[Max_domains];
 
-/* Update the sampled stats for the given domain. */
-void caml_collect_gc_stats_sample(caml_domain_state* domain)
+/* Update the sampled stats for the given domain during a STW section. */
+void caml_collect_gc_stats_sample_stw(caml_domain_state* domain)
```

Comment:
Is the change in name with the addition of the suffix `_stw` indicating that this function should only be called in stop-the-world sections? Is this true for the rest of the runtime? It may be useful to document this convention somewhere.

Reply:
Yes, I think all `foo_stw` sections are meant to be called from inside a STW section. (Of course some use the `foo_from_stw`, and some have `stw` somewhere in the name but not at the end. I am not sure where this convention should be documented; I find it relatively obvious, especially if we consistently used the less common but clearer `_from_stw` variant. We could also consider a refactoring PR to rename various functions to enforce a single version of this convention.

Reply:
I propose a PR to define and enforce naming conventions in #12619. Thanks for asking: I was in "minimal change" mode when writing new runtime code, but it actually felt nice to put these conventions on paper and ensure they are consistently followed.